tools/build-agent-release.sh

```diff
@@ -0,0 +1,54 @@
+RELEASEDIR=releases
+TAG=$(date +%Y%m%d)-0.$(git log --pretty=format:'%h' -n 1)
+
+BUILDS=(
+  # FORMAT
+  # GOOS GOARCH BINSUFFIX
+  "darwin 386 darwin-i386-$TAG.prod"
+  "darwin amd64 darwin-amd64-$TAG.prod"
+  "linux 386 linux-i386-$TAG.prod"
+  "linux amd64 linux-amd64-$TAG.prod"
+  "windows 386 windows-i386-$TAG.prod.exe"
+  "windows amd64 windows-amd64-$TAG.prod.exe"
+)
+
+
+# update_releases invokes the update_release_json tool to record an agent update.
+# It expects to be called with the following arguments:
+# 1. The path to the releases.json file to update,
+# 2. The component to update,
+# 3. The tag for the new release and
+# 4. The path to the new agent binary.
+function update_releases() {
+  releasejson=$(go run tools/update_release_json.go \
+    -releases $RELEASEDIR/$1 \
```

Comment:
These bare variables are not whitespace-safe, which is fine if none of them will ever contain spaces. Consider running `shellcheck` against this script.